### ISSUE TITLE
Global Sign Out is not working 

### ISSUE THREAD

**Issue by JeetSA** [aws-amplify/aws-sdk-android#1082]
**Describe the bug**
When we login into 2 devices and we sign out globally from one device, in other device the sign in state is true. After signing out from one device when we try to sign out globally, it is returning exception with error "Access Token has been revoked".

**Steps to Reproduce**
- Sign in into 2 devices and globally logout from one device.
- Check sign in state in other device.

**Code block**
```
AWSMobileClient.getInstance().signOut(SignOutOptions.builder().signOutGlobally(true).build(), object : Callback<Void> {
            override fun onResult(result: Void?) {
                logoutUser()
            }

            override fun onError(e: java.lang.Exception?) {
                LogUtils.e("sign-out error")
            }
        });
```

**Which AWS service(s) are affected?**
AWS Cognito

**Expected behavior**
It should sign out from all signed in devices.

**Environment Information:**
 - AWS Android SDK Version: 2.13.2
 - Device: all
 - Android Version: all
 - Specific to simulators: No

**Comment by palpatim** [comment]
This appears to be a duplicate of #1670, so the same answer applies: 

`isSignedIn()` looks at the cached state of the login on that device. If you attempt to get credentials, you should see that the refresh token is no longer valid, and the user must go through a login process on that device.

**Comment by JeetSA** [comment]
@palpatim Yes, I am getting that exception, but only after I kill the app from recent and relaunch it. Otherwise I am able to get the tokens.

**Comment by palpatim** [comment]
@JeetSA 

Let's define 2 devices: 
- Device 1: Sign out globally using the code as above:
    ```java
    AWSMobileClient.getInstance().signOut(SignOutOptions.builder().signOutGlobally(true).build(), object : Callback<Void> {
        override fun onResult(result: Void?) {
            logoutUser()
        }

        override fun onError(e: java.lang.Exception?) {
            LogUtils.e("sign-out error")
        }
    });
    ```
- Device 2: "Check sign in state"

What code are you using to check the sign in state on Device 2?

**Comment by JeetSA** [comment]
I am checking the refresh token as per your previous answer. So I am not getting exception in below function till I am not removing app from recent.
```
AWSMobileClient.getInstance().getTokens(object : Callback<Tokens> {
            override fun onResult(result: Tokens?) {
                LogUtils.e("token is ${result?.refreshToken?.tokenString}")
            }

            override fun onError(e: java.lang.Exception?) {
                LogUtils.e("onError $e")
            }

        })
```

**Comment by palpatim** [comment]
[It seems likely this is related to #1670, so I'll address this comment to both stated problems.]

The `getTokens` call retrieves its data from cached credentials unless those credentials are expired, so the behavior you're seeing is expected.

I would encourge you to reframe your question in terms of "I want to accomplish X use case" and see if detecting the global signed-in state is actually fundamental to accomplishing that use case.

It's unusual to have a use case that actually requires global sign-out state to be instantly reflected on every device on an account--the usual pattern is to let each device manage its own state, and respond to authentication errors on a per-call basis. 

In such a scenario, you would rely on the local credentials state to start a network request, the service would detect that the user has been globally signed out, and would respond with an error indicating the request is not authenticated. Such a pattern is viable for any use case, including global sign out, but also service-initiated scope and/or permissions changes.

@JeetSA, you're encountering such a pattern in your original post: making a global sign out call is an authenticated call, and the response you receive is appropriate to that call: the service can't affect global authenticated state using the provided credentials, because the provided credentials are invalid.

If your use case legitimately requires the ability to respond quickly to global sign out states, one option might be to do an authenticated "ping" that fires a network request to some authenticated endpoint, and respond to authentication errors from that request appropriately. I do not recommend making network-backed credentials calls each time; such a pattern would result in every network call being preceded by an additional call to get refreshed tokens, which would be a very poor customer experience. And again, I would encourage you to decide whether your use case really depends on having near-real-time knowledge of global sign-in state.

**Comment by palpatim** [comment]
Closing due to inactivity
